examples/python/async_streaming/README.md

```diff
@@ -0,0 +1,42 @@
+# gRPC Python Non-Blocking Streaming RPC Client Example
+
+The goal of this example is to demonstrate how to handle streaming responses without blocking the current thread.
```

Comment:
This is a really cool example of an application of bidi streaming. You probably also want to add some commentary about what this example is trying to accomplish at the application layer (i.e. in terms of phone calls.

Reply:
Added more description about the API pattern, the usage of examples, and why async streaming is needed.